src/core/stompest/sync/client.py

```diff
@@ -289,11 +289,11 @@ def canRead(self, timeout=None):
             timeout = deadline and max(0, deadline - time.time())
             if not self._transport.canRead(timeout):
                 return False
-            frame = self._transport.receive()
+            frame = self._transport.receive()            
             self.session.received()
             if self.log.isEnabledFor(logging.DEBUG):
-                self.log.debug('Received %s' % frame.info())
-            if frame:  # there's a real STOMP frame on the wire, not a heart-beat
+                self.log.debug('Received %s' % frame.info())            
+            if frame.__class__.__name__ == 'StompFrame':  # there's a real STOMP frame on the wire, not a heart-beat
```

Comment:
In Py3 interpreter StompHeartBeat frames passed through the check and caused problems, so I changed to check to be more explicit.